lib/orbit/ocs/repo/entities_server.ex

```diff
@@ -0,0 +1,154 @@
+
+  @typep output :: %{timestamp: integer(), entities: Orbit.Ocs.Entities.entities()}
+  @typep state :: %{
+           out: output(),
+           subscriptions: MapSet.t(pid()),
+           last_push_to_subscriptions: DateTime.t()
+         }
+
+  @spec start_link(any()) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(opts) do
+    Logger.info("Initialized Ocs.EntitiesServer")
+
+    # Configurable for tests
+    if Keyword.get(opts, :schedule_pushes?, true) do
+      schedule()
+    end
+
+    {:ok,
+     %{
+       out: %{timestamp: Util.Time.current_time(), entities: Orbit.Ocs.Entities.query_latest()},
+       subscriptions: MapSet.new(),
+       last_push_to_subscriptions: DateTime.utc_now()
+     }}
+  end
+
+  @impl GenServer
+  @spec handle_call({:subscribe, pid()}, GenServer.from(), state()) ::
+          {:reply, output(), state()}
+  def handle_call({:subscribe, pid}, _from, state) do
+    Process.monitor(pid)
+
+    {:reply, state.out, %{state | subscriptions: MapSet.put(state.subscriptions, pid)}}
+  end
+
+  @impl GenServer
+  @spec handle_call(
+          {:new_messages, [Message.t()]},
+          GenServer.from(),
+          state()
+        ) ::
+          {:reply, :ok, state()}
+  def handle_call({:new_messages, messages}, _from, state) do
+    apply_changes(messages)
+
+    # TODO: Avoid synchronous push here?
+    state = push_with_throttling(state)
+
+    {:reply, :ok, state}
+  end
+
+  defp schedule do
+    Process.send_after(self(), :ensure_push, 1_000)
+  end
+
+  @impl GenServer
+  def handle_info(:ensure_push, state) do
+    state = push_with_throttling(state)
+    schedule()
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  @spec handle_info({:DOWN, reference(), :process, pid(), term()}, state()) :: {:noreply, state()}
+  def handle_info({:DOWN, _monitor_ref, :process, pid, _reason}, state) do
+    {:noreply, %{state | subscriptions: MapSet.delete(state.subscriptions, pid)}}
+  end
+
+  # Subscribe
+
+  @spec subscribe(pid(), atom()) :: output()
+  def subscribe(subscriber_pid, server_name \\ __MODULE__) do
+    GenServer.call(server_name, {:subscribe, subscriber_pid})
+  end
+
+  # Push
+
+  @spec push_with_throttling(state()) :: state()
+  defp push_with_throttling(state) do
+    time_since_last_push =
+      DateTime.diff(DateTime.utc_now(), state.last_push_to_subscriptions, :second)
+
+    if time_since_last_push < @throttle_seconds do
+      Logger.debug("#{__MODULE__} : Push throttled : (#{time_since_last_push}s since last push)")
+      state
+    else
+      push(state)
+    end
+  end
+
+  @spec push(state()) :: state()
+  defp push(state) do
+    entities = Orbit.Ocs.Entities.query_latest()
+
+    state = %{
+      state
+      | out: %{
+          timestamp: Util.Time.current_time(),
+          entities: entities
+        }
+    }
+
+    Logger.info("#{__MODULE__} : Pushing to all subscribers")
```

Comment:
suggestion: same here with splunk logging

Reply:
Fixed in f6a84d2